locator/network_topology_strategy.cc

```diff
@@ -192,11 +192,11 @@ class natural_endpoints_tracker {
         : _tm(tm)
         , _tp(_tm.get_topology())
         , _dc_rep_factor(dc_rep_factor)
-        , _all_endpoints(_tp.get_datacenter_endpoints())
-        , _racks(_tp.get_datacenter_racks())
+        , _token_owners(_tm.get_datacenter_token_owners())
+        , _racks(_tm.get_datacenter_racks_token_owners())
```

Comment:
> The comment on `_racks` above has "(only with token owners)"
> 
> but in implementation of `get_datacenter_racks_token_owners` we have:
> 
> ```
>         // If there are no token owners in a DC/rack, we still want this DC/rack to appear in the result.
> ```
> 
> looks contradictory

I meant "only token owners are left in each rack", not "only racks with token owners are left". Now I see "only with token owners" is not clear and I should change it.

> also why do we want such DCs/racks to appear in the result?

Hiding a rack/DC in these functions seemed restrictive to me. Theoretically, we could have a logic that wanted to do something with a DC with no token owners. However, I doubt we will have something like this. I'll think about changing these functions to not include a DC/rack with no token owners. It would also simplify their code.

Reply:
Perhaps both approaches are fine, but whatever you choose, make sure it's clear when you look only at the function declaration (it needs to be specified as part of the interface, not an implementation detail)

Reply:
I changed these funtions o not include a DC/rack with no token owners.